snuba/query/snql/parser.py

```diff
@@ -807,6 +807,42 @@ def parse(exp: Expression) -> Expression:
     query.transform_expressions(parse)
 
 
+def _mangle_query_aliases(
+    query: Union[CompositeQuery[QueryEntity], LogicalQuery],
+) -> None:
+    """
+    If a query has a subquery, the inner query will get its aliases mangled. This is
+    a problem because the outer query is using the inner aliases, not the inner
+    selected expression values.
```

Comment:
Please explain why this is inevitable as we never execute individual subqueries and we cannot replace the column names in the response.